File: go/1032.go
package main

import (
	"fmt"
)

func main() {
	primes := getPrimes()

	for true {
		var c int

		fmt.Scanln(&c)

		if c == 0 {
			return
		}

		l := make([]int, c)

		for i := range l {
			l[i] = i + 1
		}

		pi := 0
		cp := 0

		for len(l) > 1 {
			pos := primes[pi]
			pos += cp
			pos -= 1

			if pos >= len(l) {
				pos = pos % len(l)
			}

			l = append(l[:pos], l[pos+1:]...)

			pi += 1
			cp = pos
		}

		fmt.Printf("%v\n", l[0])
	}
}

func getPrimes() []int {
	primeMap := make(map[int]bool)

	maxPrime := 40000

	primes := make([]int, 0)
	primes = append(primes, 2)
	currentPrime := 3

	for len(primes) < 3501 {
		if currentPrime%2 != 0 && !primeMap[currentPrime] {
			primes = append(primes, currentPrime)

			counter := currentPrime
			for counter <= maxPrime {
				counter += currentPrime
				primeMap[counter] = true
			}
		}

		currentPrime += 1
	}

	return primes
}

File: python/1032.py
def generate_primes():
    primes = [2]
    primes_map = {1: True, 2: True}
    max_prime = 100000

    current_prime = 3
    while len(primes) < 3501:
        if current_prime & 1 != 0 and primes_map.get(current_prime) is not False:
            primes.append(current_prime)
            counter = current_prime
            while counter < max_prime:
                counter += current_prime
                primes_map[counter] = False

        current_prime += 1

    return primes


primes = generate_primes()

while True:
    x = int(input())

    if x == 0:
        break

    if x == 1:
        print(1)
        continue

    l = []

    for i in range(x):
        l.append(i + 1)

    p = primes[:x]
    pi = 0
    cp = 0
    while len(l) > 1:
        ps = p[pi]
        ps += cp
        ps -= 1

        if ps >= len(l):
            ps = ps % len(l)

        l.pop(ps)
        pi += 1
        cp = ps

    print(l[0])
